otter/test/convergence/test_service.py

```diff
@@ -504,32 +539,61 @@ def test_filter_out_currently_converging(self):
             (Log('converge-all-groups',
                  dict(group_infos=[self.group_infos[1]],
                       currently_converging=['g1'])),
-             lambda i: None),
+             noop),
+            (ReadReference(ref=self.recently_converged), lambda i: pmap()),
+            (Func(time.time), lambda i: 100),
+            parallel_sequence([[self._expect_group_converged('01', 'g2')]])
+        ]
+        self.assertEqual(perform_sequence(sequence, eff), ['converged g2!'])
 
-            parallel_sequence([
-                [(BoundFields(mock.ANY, dict(tenant_id='01',
-                                             scaling_group_id='g2')),
-                  nested_sequence([
-                     (GetStat(path='/groups/divergent/01_g2'),
-                      lambda i: ZNodeStatStub(version=5)),
-                     (TenantScope(mock.ANY, '01'),
-                      nested_sequence([
-                         (('converge', '01', 'g2', 5, 3600),
-                          lambda i: 'converged two!'),
-                      ])),
-                  ]))],
-             ])
+    def test_filter_out_recently_converged(self):
+        """
+        If a group was recently converged, it will not be converged again.
+        """
+        eff = self._converge_all_groups(['00_g1'])
+        sequence = [
+            (ReadReference(ref=self.currently_converging), lambda i: pset([])),
+            (Log('converge-all-groups',
+                 dict(group_infos=[self.group_infos[0]],
+                      currently_converging=[])),
+             noop),
+            (ReadReference(ref=self.recently_converged),
+             lambda i: pmap({'g1': 5})),
+            (Func(time.time), lambda i: 14),
+            parallel_sequence([])  # No groups to converge
+        ]
+        self.assertEqual(perform_sequence(sequence, eff), [])
+
+    def test_dont_filter_out_non_recently_converged(self):
+        """
+        If a group was converged in the past but not recently, it will be
+        cleaned from the ``recently_converged`` map, and it will be converged.
+        """
+        eff = self._converge_all_groups(['00_g1'])
+        sequence = [
+            (ReadReference(ref=self.currently_converging), lambda i: pset([])),
+            (Log('converge-all-groups',
+                 dict(group_infos=[self.group_infos[0]],
+                      currently_converging=[])),
+             noop),
+            (ReadReference(ref=self.recently_converged),
+             lambda i: pmap({'g1': 5})),
```

Comment:
Your next test handles it - basically I just wanted to make sure that a group that was still in the `recently_converged` group that isn't being converged this time gets removed, because otherwise deleted groups will remain forever.